source/server/filter_chain_manager_impl.h

```diff
@@ -63,6 +63,7 @@ class FilterChainFactoryContextImpl : public Configuration::FilterChainFactoryCo
   envoy::config::core::v3alpha::TrafficDirection direction() const override;
   TimeSource& timeSource() override;
   ProtobufMessage::ValidationVisitor& messageValidationVisitor() override;
+  ProtobufMessage::ValidationContext& messageValidationContext() override;
```

Comment:
Why do we have both validation visitor and context as peers?

Reply:
> Why do we have both validation visitor and context as peers?

removing of validator() will be in another PR,  lots of files will be modified as I plan to pass the signal "added_via_api" (or a vlidator parameter) down from Listener to everythin.

Reply:
Trying to figure out how this works.. do you have a draft PR to see the end results? As mentioned off-line, I'd like to:
1. Thread validators around, vs. context+flag unless the flag is needed for some other purpose.
2. Ensure that it's easy for consumers to automatically "do the right thing", without having to think too hard, otherwise we get folks picking random validators via copy+paste.

^^ guided the existing design. We need a bunch of comments to explain this interim solution in any case, as it's pretty confusing why both are there right now.

Reply:
#9779 I have it now. 
That PR doesn't include the removal of FactoryContext::messageValidationVisitor() tho.
So if we want to remove that later, we  will need to pipe a picked validator from listenerImpl down to HCM::Config.